framework/src/org/checkerframework/framework/type/treeannotator/PropagationTreeAnnotator.java

```diff
@@ -170,4 +190,14 @@ public Void visitTypeCast(TypeCastTree node, AnnotatedTypeMirror type) {
         }
         return null;
     }
+
+    private boolean isFullyAnnotated(AnnotatedTypeMirror type) {
```

Comment:
I'm wondering why it is enough to only check the main modifier. For a generic type, why don't you also need to check whether the type arguments are fully annotated? Similar for other types.
Adding a comment explaining the reasoning for this would be helpful.